extensions/amp-form/0.1/amp-form.js

```diff
@@ -86,20 +90,24 @@ export class AmpForm {
 
   /** @private */
   installSubmitHandler_() {
-    this.form_.addEventListener('submit', e => this.handleSubmit_(e));
+    this.form_.addEventListener('submit', e => this.handleSubmit_(e), true);
   }
 
   /**
+   * Note on stopImmediatePropagation usage here, it is important to emulate native
+   * browser submit event blocking. Otherwise any other submit listeners would get the
+   * event.
+   *
+   * For example, action service shouldn't trigger 'submit' event if form is actually
+   * invalid. stopImmediatePropagation allows us to make sure we don't trigger it
```

Comment:
Well, it depends on registration order. With this registered in capture phase, and the action registered in bubble phase, it doesn't though.